src/test/java/org/springframework/integration/kafka/listener/AbstractBrokerTests.java

```diff
@@ -162,4 +166,30 @@ private static void sleep(int time) {
 		}
 	}
 
+	public class Sender<K,V> {
+
+		private Producer<K,V> producer;
+
+		public Sender(Producer<K, V> producer) {
+			this.producer = producer;
+		}
+
+		public void send(Collection<ProducerRecord<K,V>> records) {
+			Future<RecordMetadata> lastFuture = null;
+			for (ProducerRecord<K, V> record : records) {
+					lastFuture = producer.send(record);
+			}
+			if (lastFuture != null) {
+				try {
+					lastFuture.get();
```

Comment:
Not sure - It's a test class, really - the only goal here was to make the test asynchronous. The null check is for the case when the record list is empty, for whatever reason.